utils/src/bloom_filters/rolling_bloom_filter/cyclic_subfilter.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::VecDeque;
+
+pub struct CyclicFilter<T, const SUBFILTER_COUNT: usize> {
+    list: VecDeque<T>,
+}
```

Comment:
Is it possible/reasonable to use `GenericArray` here, since you already know the max size?

Reply:
Probably not, because I've switched to `Vec` now (to make the filter a bit faster and use less memory when it's just created). And before that I used a regular array, so `GenericArray` is not really needed here.